integration_test/images_test.go

```diff
@@ -0,0 +1,204 @@
+package integration_test
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/cavaliercoder/go-cpio"
+	diskfs "github.com/diskfs/go-diskfs"
+	"github.com/google/uuid"
+	"github.com/onsi/gomega/ghttp"
+	"github.com/openshift/assisted-image-service/internal/handlers"
+	"github.com/openshift/assisted-image-service/pkg/imagestore"
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	versions = []map[string]string{
+		{
+			"openshift_version": "pre-release",
+			"cpu_architecture":  "arm64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/arm64/dependencies/rhcos/pre-release/latest/rhcos-live.aarch64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/arm64/dependencies/rhcos/pre-release/latest/rhcos-live-rootfs.aarch64.img",
+		},
+		{
+			"openshift_version": "4.8",
+			"cpu_architecture":  "x86_64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.8/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.8/latest/rhcos-live-rootfs.x86_64.img",
+		},
+		{
+			"openshift_version": "pre-release",
+			"cpu_architecture":  "x86_64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live-rootfs.x86_64.img",
+		},
+	}
+
+	imageDir   string
+	imageStore imagestore.ImageStore
+)
+
+var _ = Describe("Image integration tests", func() {
+	var (
+		isoFilename    string
+		imageID        string
+		assistedServer *ghttp.Server
+		imageServer    *httptest.Server
+		imageClient    *http.Client
+	)
+
+	contexts := []struct {
+		name             string
+		imageType        string
+		expectedIgnition []byte
+		expectedRamdisk  []byte
+	}{
+		{
+			name:             "full-iso",
+			imageType:        imagestore.ImageTypeFull,
+			expectedIgnition: []byte("someignitioncontent"),
+			expectedRamdisk:  nil,
+		},
+		{
+			name:             "minimal-iso-with-initrd",
+			imageType:        imagestore.ImageTypeMinimal,
+			expectedIgnition: []byte("someignitioncontent"),
+			expectedRamdisk:  []byte("someramdiskcontent"),
+		},
+		{
+			name:             "minimal-iso-without-initrd",
+			imageType:        imagestore.ImageTypeMinimal,
+			expectedIgnition: []byte("someignitioncontent"),
+			expectedRamdisk:  []byte(""),
+		},
+	}
+
+	for i := range contexts {
+		ctx := contexts[i]
```

Comment:
Maybe `case`/`cases` here instead of `ctx`/`contexts` as the latter almost always reserved for a `context.Context` object.